api/src/main/java/org/apache/iceberg/types/TypeUtil.java

```diff
@@ -151,10 +152,8 @@ public static Type assignFreshIds(Type type, NextID nextId) {
    * @return a structurally identical schema with new ids assigned by the nextId function
    */
   public static Schema assignFreshIds(Schema schema, NextID nextId) {
-    return new Schema(TypeUtil
-        .visit(schema.asStruct(), new AssignFreshIds(nextId))
-        .asNestedType()
-        .fields());
+    Types.StructType struct = TypeUtil.visit(schema.asStruct(), new AssignFreshIds(nextId)).asStructType();
```

Comment:
I have another question about the `Schema#select`,  should we also project the field ids on top of `identifierFieldIds` ?

Reply:
I think `Schema#select` is mostly used for schema projection as a form of schema transformation helper method I mentioned in earlier PRs, so I think missing `identifierFieldIds` might not be a problem.

Reply:
Yes, that was why I chose to just remove the fields not matched instead of throwing NPE,. but then I realized it dies not make sense. 

For example, for a table schema with identifier `(id, catalog)`,  it does not really make sense to say that, when selecting the `id` field, the new schema has identifier `(id)`. So I don't think it the identifier fields should be added there.

Reply:
Yeah,   providing a projected identifier field ids looks very strange for people.  I think it's okay to remove the identifier fields when projecting.